libs/input-mapping/src/Helper/FakeDevStorageTableRewriteHelper.php

```diff
@@ -10,37 +10,38 @@
 use Keboola\StorageApiBranch\ClientWrapper;
 use Psr\Log\LoggerInterface;
 
-class SourceRewriteHelper
+class FakeDevStorageTableRewriteHelper implements TableRewriteHelperInterface
 {
-    public static function rewriteTableOptionsSources(
+    public function rewriteTableOptionsSources(
         InputTableOptionsList $tablesDefinition,
         ClientWrapper $clientWrapper,
         LoggerInterface $logger
     ): InputTableOptionsList {
-        if ($clientWrapper->hasBranch()) {
-            foreach ($tablesDefinition->getTables() as $tableOptions) {
-                $tableOptions->setSource(self::rewriteSource($tableOptions->getSource(), $clientWrapper, $logger));
+        foreach ($tablesDefinition->getTables() as $tableOptions) {
+            if ($clientWrapper->hasBranch()) {
+                $tableOptions->setSource($this->rewriteSource($tableOptions->getSource(), $clientWrapper, $logger));
             }
+            $tableOptions->setSourceBranchId($clientWrapper->getDefaultBranch()['branchId']);
```

Comment:
Source branchId zapíšeme vždycky, aby se pak dal na connection zrušit fallback a bylo to explicitní.
Pro fake storage je sourceBranch vždycky produkce

Reply:
Nerozumím otázce

Reply:
Cize kdyz to vemu se zhora, budu se trochu hloupe ptat, sorry :)

- zalozim si novou dev branch - mam stejny konfigurace jako v prod, tzn stejny IM
- v tehle branchi nemam zadnou tabulku, tzn kdyz pustim nejaky job s IM, tak ten pouzjie normalne produkcni tabulku, protoze v dev branchi neexistuje tabulka se stejnym nazvem

- pak druhy pripad je, ze mam tu dev branch a zalozim si v ni nejakou novou tabulku a tu strcim do nejake nove konfigurace do IM
- je ta branch uz ulozena v konfiguraci toho IM? Nebo ten konfig IM vypada uplne stejne jako by to bylo v produkci?

Reply:
Branch není uložená nikde, konfig vypadá úplně stejně jako v produkci. To tak bylo i teď. Která tabulka se použije se rozhoduje až při spuštění a to jen podle toho jestli ta tabulka v branchi existuje nebo ne. Pokud existuje, tak se použije dev tabulka, která se ale v tomhle FakeDevStorage jmenuje jinak - to je o řádek výš a proto se nachází vždycky v default branchi (to je řádek 24)

Reply:
Diky za vysvetleni, ja sem to chtel napsat k te RealDevStorage, ale to je jedno